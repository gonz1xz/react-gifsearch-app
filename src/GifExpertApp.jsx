import { useState } from 'react';
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

	const [categories, setCategories] = useState([]);

	const onAddCategory = (newCategory) => {

		if (categories.includes(newCategory)) return console.log("Ya existe");
		setCategories([newCategory, ...categories]);
	}


	return (
		<>
			<h1>Search Gifs App</h1>

			<AddCategory
				onNewCategory={onAddCategory}
			/>

			{
				categories.map((category) => (
					<GifGrid key={category} category={category} />
				))
			}

		</>
	)
}