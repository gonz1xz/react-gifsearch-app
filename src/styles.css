* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(216, 216, 216);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(0, 60, 255);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h1 {
  text-align: center;
  padding: 0 0 10px 0;
}

h2 {
  font-size: 1.2rem;
}

h3 {
  font-size: 1.5rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  position: relative;
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  width: 300px;
  height: 300px;  
  justify-content: center;
  margin: 20px 20px 20px 0;
  overflow: hidden;
}

.card img {
  object-fit: cover;
  width: 100%;
  height: 100%;
}

.card p {
  position: absolute;
  background-color: rgba(0, 0, 0, 0.479);
  color: white;
  width: 100%;
  height: 100%;
  top: 85%;
  left: 0;
  margin: 0 auto;
  text-align: center;
}