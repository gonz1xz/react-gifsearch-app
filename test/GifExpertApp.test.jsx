import { render, screen, fireEvent } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

/*
Escribir en el input
Postear el formulario
Snapshop
Si envio misma categoria
Si envio categoria distinta
*/

describe('Pruebas en GifExpertApp', () => {

    test('Debe cambiar el valor del input ', () => {

        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');

        fireEvent.input(input, { target: { value: "Cars" } });

        screen.debug();

        expect(input.value).toBe('Cars');

    });

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();

    });

    test('Si envio misma categoria no realiza ninguna accion', () => {

    });

    test('Si envio otra categoria se agrega al array', () => {

    });




});
