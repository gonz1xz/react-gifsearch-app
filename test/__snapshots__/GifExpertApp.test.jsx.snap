// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Search Gifs App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
